proxy/http/HttpSM.cc

```diff
@@ -6626,6 +6630,9 @@ HttpSM::setup_blind_tunnel(bool send_response_hdr)
   milestones[TS_MILESTONE_SERVER_BEGIN_WRITE] = Thread::get_hrtime();
   if (send_response_hdr) {
     client_response_hdr_bytes = write_response_header_into_buffer(&t_state.hdr_info.client_response, to_ua_buf);
+    if (initial && initial->read_avail()) {
```

Comment:
For `CONNECT` method, I think we also need to transfer the data(in server_reader_buffer) after sending response header. So, could we move it out of block and remove the changes in handle_api_return ?

Finally do not forget to consume data just like HttpSM::server_transfer_init